### PR TITLE
add intensity to AmbientLight.

### PR DESCRIPTION
**Pull request by bhouston**
Implementation of improvement discussed here: https://github.com/mrdoob/three.js/issues/7620

### CODE REVIEW

**Comment by bhouston** [comment]
@WestLangley this PR makes sense?  I think it is pretty straightforward.  I've done a first pass at adding the units to the docs as well, which I think we are in agreement except for directional light (which we can fix once we figure out the right answer.)

File: docs/api/core/Object3D.html [hunk 1]
```diff
@@ -143,7 +143,7 @@
 
 		<h3>[property:Matrix4 matrixWorld]</h3>
 		<div>
-		The global transform of the object. If the Object3d has no parent, then it's identical to the local transform.
+		The global transform of the object. If the Object3d has no parent, then it's identical to the local transform.  World transforms are assumed to be specified in meters.
```
Comment (WestLangley) [inline docs/api/core/Object3D.html]: I'm not sure I would say that a matrix transform has units of meters.

**Comment by WestLangley** [comment]
> this PR makes sense? 

Yes.

Except the docs. I don't think I am quite as confident in what the units will be as you are --  but I guess we can always change the docs later...

Do we agree that color is a hue and can be considered as unit-less, and the units come from elsewhere? Or do you disagree with me on that hypothesis?

**Comment by bhouston** [comment]
> Do we agree that color is a hue and can be considered as unit-less, and the units come from elsewhere? Or do you disagree with me on that hypothesis?

I think that because our color is restricted to LDR, it can not represent unbounded colors required for lighting or emission, although it is great for relative spectral measures like absorption, and reflection.  But the decomposition of a HDR light color into color and intensity is not well defined unlike the decomposition of a line into a unitless direction and length.  The reason is that we do not use "normalized colors" of some type of unit intensity.

Because we do not use a normalized color, the decomposition of a HDR light color is arbitrary.  Remember:

RGB( 0.1, 0.1, 0.1 ) * intensity( 10 ) === RBG( 1, 1, 1 ) * intensity( 1 )

Because these are equivalent in result, but their intensities different by an order of magnitude, it is hard to say with any accuracy that the units are in intensity.

We could explore normalizing our color, but I'd prefer to find a precedent for this in another tool, and I worry about being confusing to users used to how other tools work.

**Comment by mrdoob** [comment]
Ended up implementing it in a different way. Better to do another PR with the metric/lumens stuff.

I think I want to fix the shadows code first, then release r74 and then we can proceed with the lights changes.

**Comment by bhouston** [comment]
Wohoo!  Thanks for getting the AmbientLight intensity in. :)

**Comment by mrdoob** [comment]
Thank you!

**Pull request closed without merging**